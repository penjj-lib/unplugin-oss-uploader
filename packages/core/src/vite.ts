import type { UserOptions } from './core/plugin'
import { plugin } from './core/plugin'

export default plugin.vite as (opts: UserOptions) => any
