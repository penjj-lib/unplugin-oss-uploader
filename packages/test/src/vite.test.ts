import { resolve } from 'node:path'
import { readFile, rm } from 'node:fs/promises'
import { build } from 'vite'
import { afterAll, expect, it } from 'vitest'
import vue from '@vitejs/plugin-vue'
import cdnUploader from 'unplugin-oss-uploader/vite'
import { clearPath, mockedAdapter } from './testUtils'

const root = resolve(process.cwd(), 'src/fixture')
const resolveRoot = (...path: string[]) => resolve(root, ...path)

afterAll(() => {
  rm(resolveRoot('./dist'), { recursive: true, force: true })
})

it('vite build', async () => {
  await build({
    root,
    logLevel: 'silent',
    plugins: [
      vue(),
      cdnUploader({
        folderName: 'vite',
        adapter: mockedAdapter(),
      }),
    ],
    resolve: {
      alias: {
        '@': '.',
      },
    },
    build: {
      minify: false,
      outDir: 'dist',
      assetsInlineLimit: 0,
      rollupOptions: {
        external: ['vue'],
        output: {
          preserveModules: false,
          entryFileNames: () => 'vite.js',
          assetFileNames: info => info.name!,
        },
      },
    },
  })

  const js = await readFile(resolveRoot('./dist/vite.js'), 'utf8')
  expect(clearPath(js)).toMatchSnapshot()

  const css = await readFile(resolveRoot('./dist/index.css'), 'utf8')
  expect(clearPath(css)).toMatchSnapshot()
})
