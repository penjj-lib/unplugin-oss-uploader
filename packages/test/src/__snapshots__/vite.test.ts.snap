// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vite build 1`] = `
"import { openBlock, createElementBlock, Fragment, createElementVNode, toDisplayString, unref } from "vue";
(function polyfill() {
  const relList = document.createElement("link").relList;
  if (relList && relList.supports && relList.supports("modulepreload")) {
    return;
  }
  for (const link of document.querySelectorAll('link[rel="modulepreload"]')) {
    processPreload(link);
  }
  new MutationObserver((mutations) => {
    for (const mutation of mutations) {
      if (mutation.type !== "childList") {
        continue;
      }
      for (const node of mutation.addedNodes) {
        if (node.tagName === "LINK" && node.rel === "modulepreload")
          processPreload(node);
      }
    }
  }).observe(document, { childList: true, subtree: true });
  function getFetchOpts(link) {
    const fetchOpts = {};
    if (link.integrity)
      fetchOpts.integrity = link.integrity;
    if (link.referrerPolicy)
      fetchOpts.referrerPolicy = link.referrerPolicy;
    if (link.crossOrigin === "use-credentials")
      fetchOpts.credentials = "include";
    else if (link.crossOrigin === "anonymous")
      fetchOpts.credentials = "omit";
    else
      fetchOpts.credentials = "same-origin";
    return fetchOpts;
  }
  function processPreload(link) {
    if (link.ep)
      return;
    link.ep = true;
    const fetchOpts = getFetchOpts(link);
    fetch(link.href, fetchOpts);
  }
})();
const vite$1 = "https://example.com/vite/acee37c25f1f47dd56cc29c49190fb9e.png";
const vite = "https://example.com/vite/acee37c25f1f47dd56cc29c49190fb9e.png";
const _export_sfc = (sfc, props) => {
  const target = sfc.__vccOpts || sfc;
  for (const [key, val] of props) {
    target[key] = val;
  }
  return target;
};
const _hoisted_1 = /* @__PURE__ */ createElementVNode(
  "img",
  { src: vite$1 },
  null,
  -1
  /* HOISTED */
);
const _sfc_main = {
  __name: "DemoComp",
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(
        Fragment,
        null,
        [
          _hoisted_1,
          createElementVNode(
            "div",
            null,
            toDisplayString(unref(vite)),
            1
            /* TEXT */
          )
        ],
        64
        /* STABLE_FRAGMENT */
      );
    };
  }
};
const DemoComp = /* @__PURE__ */ _export_sfc(_sfc_main, [["__file", "/src/fixture/DemoComp.vue"]]);
console.log(vite$1, DemoComp);
"
`;

exports[`vite build 2`] = `
".box1 {
  background: url('https://example.com/vite/aa5b4c6d7f2c4f7327ba4add2e8782bf.png');
}
.box2 {
  background: url('https://example.com/imgs/webpack.png');
}
@font-face {
  font-family: "Test";
  src: url("https://example.com/vite/d41d8cd98f00b204e9800998ecf8427e.ttf");
}
div {
  background: url("https://example.com/vite/aa5b4c6d7f2c4f7327ba4add2e8782bf.png");
}
.box1 {
  background: url('https://example.com/vite/aa5b4c6d7f2c4f7327ba4add2e8782bf.png');
}
.box2 {
  background: url('https://example.com/imgs/webpack.png');
}
"
`;
